src/averageAges.js

```diff
@@ -1,61 +1,56 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter((
+    {
+      died,
+      sex,
+    }) => century ? sex === 'm'
+  && Math.ceil(died / 100) === century : sex === 'm');
+
+  return men.reduce((a, b) => a + b.died - b.born, 0) / men.length;
 }
 
 /**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
  * @param {object[]} people
  * @param {boolean} withChildren - optional
  *
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter((
+    {
+      name,
+      sex,
+    }) => withChildren ? sex === 'f'
+  && people.some(kid => kid.mother === name) : sex === 'f');
+
+  return women.reduce((a, b) => a + b.died - b.born, 0)
+    / women.length;
 }
 
 /**
- * Implement calculateAverageAgeDiff function.
- *
- * The function returns an average age difference between a child and his or her
- * mother in the array. (A mother's age at child birth)
- *
- * If `onlyWithSon` is specified then function calculates age difference only
- * for sons and their mothers.
- *
  * @param {object[]} people
  * @param {boolean} onlyWithSon - optional
  *
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childWithMother = people.filter((
+    {
+      sex,
+      mother,
+    }) => onlyWithSon ? sex === 'm' && mother !== null
```

Comment:
no need to write `&& mother !== null`, could be just `&& mother`

In JavaScript, the `!==` operator checks for strict inequality, meaning it checks both the value and the type of the operands. When comparing a variable to null, the `!==` operator ensures that the variable is not only not equal to `null` in value but also not of the type `null`.

However, since `null` is a falsy value in JavaScript, you can simply use something in a conditional statement to check if it evaluates to a truthy value. If something is not `null`, it will be considered truthy and pass the conditional check.

So, if you only need to check if a value is truthy (i.e., not null, undefined, false, 0, an empty string, or NaN), you can use `if (something)` instead of `if (something !== null)`. It simplifies the code and conveys the same meaning.